Redirect bug

bdc34 opened the pull request:
A GET to /login should redirect to /user when the client is already logged in. This works but it ignores any next_page query param the client sends.

I've included a fix and a unit test for this in this PR. 

The unit test doesn't work in spite of being almost identical to other unit tests in the suite. Send me any advice or ideas you might have about this. I'm running the unit test with

`WITH_INTEGRATION=1 pipenv run pytest 
    --cov=accounts 
    --cov=users/arxiv 
    --cov-report=term-missing 
    accounts users/arxiv`

bmaltzan commented on accounts/accounts/tests/test_end_to_end.py:
```diff
@@ -415,6 +419,22 @@ def test_post_login(self):
+        response = client.post('/login', data=form_data)
+        self.assertEqual(response.status_code, status.HTTP_303_SEE_OTHER)
+
+        next_page = 'https://arxiv.org/some_sort_of_next_page?cheeseburger=yes%20please'
```
Is the domain ok to include in next_page, in the client.get call?

bmaltzan replied:
In routes/ui.py#anonymous_only#wrapper
request.auth is None, when called by the test's client.get.
Maybe it needs to be adjusted to send cookies?

bdc34 replied:
> 

It does seem like a danger. 
It is critical to be able to go to URLs like https://mod.arxiv.org/something or services.dev.arxiv.org. 
We could restrict it to just subdomains of arxiv.org or have a regex.

bdc34 replied:
Re: request.auth is None, when called by the test's client.get. 

I checked that the cookies are sent and they seem to be there on the server. reqeust.auth gets set by code in arxiv-auth/user and I'm wondering if that is not setup in the tests.

bdc34 replied:
@bmaltzan Thanks again, you advice was helpful.

bmaltzan commented:
These args reduce the output a little:
pytest -k test_get_login  accounts/accounts/tests/test_end_to_end.py

bdc34 commented:
I've added an end to end test that logs in and actually tries to get a protected endpoint. 

I looked through all the tests in arxiv-auth and didn't find this kind of test. It seems critical so I added it.

It is test_login_get_something_protected() in accounts/accounts/tests/test_end_to_end.py

bdc34 commented:
The test pass on my laptop but this is failing in travis with: https://app.travis-ci.com/github/arXiv/arxiv-auth/builds/243063631
